mutation/mutation_partition_serializer.cc

```diff
@@ -7,6 +7,8 @@
  * SPDX-License-Identifier: AGPL-3.0-or-later
  */
 
+#include <seastar/core/preempt.hh>
+
```

Comment:
> mutation_partition_serializer: add write_in_thread
> For stall-free freezing in a thread.

Why a thread?

Reply:
because the `ser::writer_of_mutation<bytes_ostream>` generated serialization code is synchronous.
See for example freeze_in_thread:
```
    auto fm = frozen_mutation(m.key());
    mutation_partition_serializer part_ser(*m.schema(), m.partition());

    ser::writer_of_mutation<bytes_ostream> wom(fm._bytes);
    std::move(wom).write_table_id(m.schema()->id())
                .write_schema_version(m.schema()->version())
                .write_key(m.key())
                .partition([&] (auto wr) {
                    part_ser.write_in_thread(std::move(wr));
                }).end_mutation();
```

Maybe we could generate async serialization boilerplate code in addition to the synchronous entry points
and then we would not need to run this in a thread.

If we had async serialization boiler

Reply:
Let's generate async serialization code then, otherwise we throw threads everywhere